app/src/main/java/es/jvbabi/vplanplus/data/repository/BaseDataRepositoryImpl.kt

```diff
@@ -62,35 +63,28 @@ class BaseDataRepositoryImpl(
         }
     }
 
+    @OptIn(ExperimentalEncodingApi::class)
     override suspend fun getFullBaseData(
         schoolId: Long,
         username: String,
         password: String
     ): DataResponse<XmlBaseData?> {
-        val onlineRequest = OnlineRequest(logRecordRepository)
-        val classesResponse = onlineRequest.getResponse(
-            "https://www.stundenplan24.de/$schoolId/wplan/wdatenk/SPlanKl_Basis.xml",
-            username,
-            password
+        sp24NetworkRepository.authentication = BasicAuthentication(username, password)
+        val classesResponse = sp24NetworkRepository.doRequest(
+            "/$schoolId/wplan/wdatenk/SPlanKl_Basis.xml"
         )
-        val teachersResponse = onlineRequest.getResponse(
+        val teachersResponse = sp24NetworkRepository.doRequest(
             "https://www.stundenplan24.de/$schoolId/wplan/wdatenl/SPlanLe_Basis.xml",
```

Comment:
This won't work since the domain-name is already set in sp24networkRepository. The first string argument in doRequest is a _path_, not a full URL

Reply:
511dd472eb33d8951946493182b1a4a3d90045e8